source/MaterialXGenGlsl/Nodes/BitangentNodeGlsl.cpp

```diff
@@ -47,12 +48,23 @@ void BitangentNodeGlsl::emitFunctionCall(const ShaderNode& node, GenContext& con
         const string prefix = shadergen.getVertexDataPrefix(vertexData);
         if (space == WORLD_SPACE)
         {
+            ShaderPort* normal = vertexData[HW::T_NORMAL_WORLD];
```

Comment:
This can cause the emitted code to be unpredicable depending on whether the normals and or the tangents have already been emitted. 

I guess this computation change is also spawns from the fact that bitangents are not bound precomputed streams and thus are computed here.

Reply:
Can you elaborate on this?

I understand that the normal already might have been emitted in SurfaceNodeGlsl:
https://github.com/AcademySoftwareFoundation/MaterialX/blob/46de5f9153aec01243fce24590b15fd71da89f8e/source/MaterialXGenGlsl/Nodes/SurfaceNodeGlsl.cpp#L81-L86

However, that's why the `if (!normal->isEmitted())` check exists here. And the calculation is the same. Also, only `T_IN_NORMAL/TANGENT` is used in the calculation which I expect to exist?